src/Datasource/EntityTrait.php

```diff
@@ -499,17 +499,33 @@ public function offsetUnset($offset)
     }
 
     /**
-     * Determines whether a method exists in this class
+     * Fetch accessor method name
+     * Accessor methods (available or not) are cached in $_accessors
      *
-     * @param string $method the method to check for existence
-     * @return bool true if method exists
+     * @param string $property the field name to derive getter name from
+     * @param string $type the accessor type ('get' or 'set')
+     * @return string method name or empty string (no method available)
      */
-    protected function _methodExists($method)
+    protected function _accessor($property, $type = 'get')
     {
-        if (empty(static::$_accessors[$this->_className])) {
-            static::$_accessors[$this->_className] = array_flip(get_class_methods($this));
+        if (!isset(static::$_accessors[$this->_className][$type][$property])) {
+            /* first time for this class: build all fields */
+            if (empty(static::$_accessors[$this->_className])) {
+                foreach (get_class_methods($this) as $m) {
+                    $t = substr($m, 1, 3);
+                    if ($m[0] === '_' && in_array($t, ['get', 'set'])) {
+                        $p = Inflector::underscore(substr($m, 4));
```

Comment:
I chose the one character names to avoid clashing with the outer part of the method (argument name $property). Before I had $method, $type, and $prop. I think it is a bit more dangerous for mixing $prop/$property up accidentially. What do you think?